File: app/src/main/java/br/com/levimendesestudos/spidermagazine/model/Revista.java
package br.com.levimendesestudos.spidermagazine.model;

import android.os.Parcel;
import android.os.Parcelable;

/**
 *
 * Created by 809778 on 21/12/2016.
 *
 */
public class Revista implements Parcelable {

    public int id;
    public String description;
    public String thumbnailPath;
    public int issueNumber;
    public String title;
    public String date;
    public double price;
    public int pageCount;

    public Revista() {}

    protected Revista(Parcel in) {
        id = in.readInt();
        description = in.readString();
        thumbnailPath = in.readString();
        issueNumber = in.readInt();
        title = in.readString();
        date = in.readString();
        price = in.readDouble();
        pageCount = in.readInt();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(id);
        dest.writeString(description);
        dest.writeString(thumbnailPath);
        dest.writeInt(issueNumber);
        dest.writeString(title);
        dest.writeString(date);
        dest.writeDouble(price);
        dest.writeInt(pageCount);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Revista> CREATOR = new Creator<Revista>() {
        @Override
        public Revista createFromParcel(Parcel in) {
            return new Revista(in);
        }

        @Override
        public Revista[] newArray(int size) {
            return new Revista[size];
        }
    };
}

File: app/src/main/java/br/com/levimendesestudos/spidermagazine/model/Revista.kt
package br.com.levimendesestudos.spidermagazine.model

import android.os.Parcel
import android.os.Parcelable

/**
 *
 * Created by 809778 on 21/12/2016.
 *
 */
class Revista : Parcelable {

    var id: Int = 0
    var description: String? = null
    var thumbnailPath: String? = null
    var issueNumber: Int = 0
    var title: String? = null
    var date: String? = null
    var price: Double = 0.toDouble()
    var pageCount: Int = 0

    constructor()

    protected constructor(parcel: Parcel) {
        id = parcel.readInt()
        description = parcel.readString()
        thumbnailPath = parcel.readString()
        issueNumber = parcel.readInt()
        title = parcel.readString()
        date = parcel.readString()
        price = parcel.readDouble()
        pageCount = parcel.readInt()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(id)
        dest.writeString(description)
        dest.writeString(thumbnailPath)
        dest.writeInt(issueNumber)
        dest.writeString(title)
        dest.writeString(date)
        dest.writeDouble(price)
        dest.writeInt(pageCount)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<Revista> = object : Parcelable.Creator<Revista> {
            override fun createFromParcel(parcel: Parcel): Revista {
                return Revista(parcel)
            }

            override fun newArray(size: Int): Array<Revista?> {
                return arrayOfNulls(size)
            }
        }
    }
}